#!/usr/bin/env python
PACKAGE = "fyp"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("take_off",                         bool_t,   0, "Take off Drone",  False)
gen.add("land",                             bool_t,   0, "Land Drone",  False)
gen.add("forward",                          bool_t,   0, "Move Forward",  False)
gen.add("backward",                         bool_t,   0, "Move Backward",  False)
gen.add("left",                             bool_t,   0, "Move Left",  False)
gen.add("right",                            bool_t,   0, "Move Right",  False)
gen.add("hover",                            bool_t,   0, "Halt",  False)
gen.add("deleteWaypoints",                  bool_t,   0, "Delete waypoints",  False)
gen.add("followFlightPathWaypoints",        bool_t,   0, "Follow Flight Path Waypoints",  False)
gen.add("followFlightPathDwm1001",          bool_t,   0, "Follow Flight Path DWM1001",  False)


gen.add("clear_WP",         bool_t,   0, "Clear WP",  False)
gen.add("go_to_waypoint",           bool_t,   0, "Go to point",  False)
gen.add("look_and_go",              bool_t,   0, "Look and Go",  False)
gen.add("load_waypoint_gazebo",     bool_t,   0, "Load waypoint for gazebo",  False)
gen.add("load_waypoint_dwm1001",    bool_t,   0, "Load waypoint for gazebo",  False)





gen.add("targetInMapX",             double_t, 0, "target x", 0, -10,   10)
gen.add("targetInMapY",             double_t, 0, "target y", 0, -10,   10)
gen.add("targetInMapZ",             double_t, 0, "target z", 0, -10,   10)

exit(gen.generate(PACKAGE, "fyp", "droneGUI"))